plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -19,58 +19,66 @@
  * @return array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string}|WP_Error Array of plugin data or WP_Error if failed.
  */
 function perflab_query_plugin_info( string $plugin_slug ) {
-	$plugin = get_transient( 'perflab_plugin_info_' . $plugin_slug );
-
-	if ( is_array( $plugin ) ) {
-		/**
-		 * Validated (mostly) plugin data.
-		 *
-		 * @var array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string} $plugin
-		 */
-		return $plugin;
+	$transient_key = 'perflab_plugins_info';
+	$plugins       = get_transient( $transient_key );
+
+	if ( is_array( $plugins ) && isset( $plugins[ $plugin_slug ] ) ) {
+		return $plugins[ $plugin_slug ];
 	}
 
-	$fields = array(
-		'name',
-		'slug',
-		'short_description',
-		'requires',
-		'requires_php',
-		'requires_plugins',
-		'download_link',
-		'version', // Needed by install_plugin_install_status().
-	);
+	$response = wp_remote_get( 'https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[author]=wordpressdotorg&request[tag]=performance&request[per_page]=100' );
 
-	$plugin = plugins_api(
-		'plugin_information',
-		array(
-			'slug'   => $plugin_slug,
-			'fields' => array_fill_keys( $fields, true ),
-		)
-	);
-
-	if ( is_wp_error( $plugin ) ) {
-		return $plugin;
+	if ( is_wp_error( $response ) ) {
+		return new WP_Error( 'api_error', __( 'Failed to retrieve plugins data from WordPress.org API: ' . $request->get_error_message(), 'performance-lab' ) );
```

Comment:
The translation string cannot have any dynamic component:

```suggestion
		return new WP_Error( 'api_error', __( 'Failed to retrieve plugins data from WordPress.org API: ', 'performance-lab' ) . $request->get_error_message() );
```

Reply:
Even better would be to use interpolation. Not a good practice to end translation strings in whitespace.

```suggestion
		return new WP_Error(
			'api_error',
			sprintf(
				/* translators: %s: API error message */
				__( 'Failed to retrieve plugins data from WordPress.org API: %s', 'performance-lab' ),
				$request->get_error_message()
			)
		);
```